### ISSUE TITLE
weechat not loading python plugin (Error: unable to load plugin python.so ... undefined symbol: _Py_NoneStruct)

### ISSUE THREAD

**Issue by TRSx80** [weechat/weechat#1458]
Hallo. First just want to thank everyone for all their work, I have really been enjoying weechat for maybe 1 year now and it has been pretty bullet-proof IMO, so thanks for that! :smile: 

## Bug summary
I have been using [weechat-notify-send](https://github.com/s3rvac/weechat-notify-send) to send desktop notifications, which was working until recently. N.B.: that is Python notify-send and is not the one in Weechat plug-ins repository, the one in repository is is a Perl script IIRC.

So recently I noticed I stopped receiving notifications from weechat (yet I am still receiving notifications from other programs) and so I went investigating today.

I start looking at loaded plug-ins and realize that `notify-send` is not loaded, and neither is the `python` plug-in:
```
│14:38:30 =!= | Error: unable to load plugin "/usr/lib/x86_64-linux-gnu/weechat/plugins/python.so": /usr/lib/x86_64-linux-gnu/weechat/plugins/python.so: undefined symbol: _Py_NoneStruct
│14:38:30 =!= | If you're trying to load a script and not a C plugin, try command to load scripts (/perl, /python, ...)
```

## Steps to reproduce

1. Start `weechat` from command line: `$ weechat <enter>`.

## Current behavior
Python plug-in seems not to load, so then `notify-send` (which depends on it) does not load, either.

## Expected behavior
Python and `notify-send` plug-ins should load on start up automatically, as they are both set up to do.

## Suggested solutions
I am not sure.

## Additional information
I am only a low level wizard, but my searching Internet so far today led me to [this](https://bugzilla.redhat.com/show_bug.cgi?id=1656633) RedHat bug report. However when I look at the current version of [FindPython.cmake](https://github.com/weechat/weechat/blob/master/cmake/FindPython.cmake) it looks completely different. Also from reading some PRs it seems this was re-worked at some point. So I figured I'm getting in over my head now and best stop and seek help.

What changed since it was working? Well, I have been doing lots of changes to my desktop environment lately, I installed `i3-gaps`, `dunst`, `picom` (`compton` fork), etc... (some of those from source). Then I also installed [`py3status`](https://github.com/ultrabug/py3status) (drop-in replacement for `i3status`) recently via `pip3` (via `--user`, which is now default on Debian), maybe that broke something? So this may or may not be reproducible, I am really not sure what to do at this point, other than file this bug report and seek further guidance. Any of which would be greatly appreciated, of course.

Cheers! :beers: 

## MANDATORY INFO:

- WeeChat version: `2.6-2`
- OS, distribution and version: `Debian Testing`

**Comment by flashcode** [comment]
Hi,
You should try to compile latest version, 2.7.1.
This error is maybe because you compile with some Python version and link/execute with another version.

**Comment by TRSx80** [comment]
Thanks flashcode for such quick feedback.

If I am recalling correctly, I am pretty sure I installed from Debian (Testing) repos, as opposed to compiling from source.

Having said that, I did also do `apt update` today, and I see that (amongst others) `weechat` is available for upgrade to `2.7.1-1` in Debian Testing:
```
weechat-core/testing 2.7.1-1 amd64 [upgradable from: 2.6-2+b2]
weechat-curses/testing 2.7.1-1 amd64 [upgradable from: 2.6-2+b2]
weechat-perl/testing 2.7.1-1 amd64 [upgradable from: 2.6-2+b2]
weechat-plugins/testing 2.7.1-1 amd64 [upgradable from: 2.6-2+b2]
weechat-python/testing 2.7.1-1 amd64 [upgradable from: 2.6-2+b2]
weechat-ruby/testing 2.7.1-1 amd64 [upgradable from: 2.6-2+b2]
weechat/testing,testing 2.7.1-1 all [upgradable from: 2.6-2]
```
However I am simultaneously trying to help track down a [bug](https://github.com/yshui/picom/issues/329) in `picom` and I am not sure if I should keep my overall environment stable in the meantime or not (maybe that doesn't matter). There are apparently 273 packages available to be upgraded in Debian Testing for me at the moment.

**Comment by TRSx80** [comment]
Yeah, I decided just to `apt update` and that took care of the problem.

Thanks again! :beers:
